### PR TITLE
continue update stream screen  , add search feature

### PR DESCRIPTION
**Pull request by hieuruoi7797**

### CODE REVIEW

File: app/src/main/java/com/example/admin/scloud/screen/main/MainContract.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import com.example.admin.scloud.BasePresenter;
+import com.example.admin.scloud.BaseView;
+
+public interface MainContract {
+    interface View extends BaseView<Presenter> {
+        void showTabLayout();
+
+        void hideTabLayout();
+
+
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/scloud/screen/main/MainContract.java]: remove bl and format code

File: app/src/main/java/com/example/admin/scloud/screen/search/SearchContract.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.example.admin.scloud.BasePresenter;
+import com.example.admin.scloud.BaseView;
+import com.example.admin.scloud.data.model.Track;
+
+import java.util.ArrayList;
+
+public interface SearchContract {
+    interface Presenter extends BasePresenter {
+        void searchTracks(String name,int limit, int offset);
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/scloud/screen/search/SearchContract.java]: format code

File: app/src/main/java/com/example/admin/scloud/screen/search/SearchFragment.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    @Override
+    public void showLoadingTrackError(String err) {
+
+    }
+
+    @Override
+    public void setPresenter(SearchContract.Presenter presenter) {
+        this.mPresenter = presenter;
+
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/scloud/screen/search/SearchFragment.java]: remove bl

File: app/src/main/java/com/example/admin/scloud/screen/search/SearchFragment.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    @Override
+    public void showLoadingIndicator() {
+        mProgressLoading.setVisibility(View.VISIBLE);
+    }
+
+    @Override
+    public void hideLoadingIndicator() {
+        mProgressLoading.setVisibility(View.GONE);
+
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/scloud/screen/search/SearchFragment.java]: remove bl

File: app/src/main/java/com/example/admin/scloud/screen/search/SearchFragment.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    @Override
+    public void showLoadingTrackError(String err) {
+
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/scloud/screen/search/SearchFragment.java]: //TODO

File: app/src/main/java/com/example/admin/scloud/data/source/remote/BaseFetchTrackFromUrl.java [hunk 6]
```diff
@@ -107,16 +105,18 @@ protected Track parseJsonObjectToTrackObject(JSONObject jsonTrack) {
...
         track.setId(trackJSON.optInt(Track.TrackEntity.ID));
         track.setTitle(trackJSON.optString(Track.TrackEntity.TITLE));
         track.setLikeCount(trackJSON.optInt(Track.TrackEntity.LIKES_COUNT));
+
         if (jsonUser != null) {
             track.setArtist(jsonUser.optString(Track.TrackEntity.USERNAME));
         }
         return track;
     }
+
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/admin/scloud/data/source/remote/BaseFetchTrackFromUrl.java]: remove bl

**Pull request merged**